WordPress/src/main/java/org/wordpress/android/util/audio/IAudioRecorder.kt

```diff
@@ -0,0 +1,22 @@
+package org.wordpress.android.util.audio
+
+import android.Manifest
+import kotlinx.coroutines.flow.Flow
+
+interface IAudioRecorder {
+    fun startRecording(onRecordingFinished: (String) -> Unit)
+    fun stopRecording()
+    fun pauseRecording()
+    fun resumeRecording()
+    fun recordingUpdates(): Flow<RecordingUpdate>
+    fun setRecordingParams(params: RecordingParams)
+
+    companion object {
+        val REQUIRED_RECORDING_PERMISSIONS = arrayOf(
+            Manifest.permission.RECORD_AUDIO,
+            // todo pantelis: do we need this?
+            // Manifest.permission.WRITE_EXTERNAL_STORAGE
```

Comment:
@pantstamp - I don't think we need WRITE_EXTERNAL_STORAGE permission since you are using getCacheDir or getExternalFilesDir.

If we intend to open AudioRecorder to save files elsewhere we may need different permissions, but then again, that should be part of the File creation part. Just food for thought

Reply:
Removed it. @zwarm is there a scenario that you can think of that we might actually need to save files elsewhere?

Reply:
@pantstamp - Nope, I can't think of an immediate scenario that we might need to save files. If it comes up as a request in the future, then we can address it. 👍